runtime/sema/simple_type.go

```diff
@@ -37,6 +37,7 @@ type SimpleType struct {
 	Name                 string
 	QualifiedName        string
 	TypeID               TypeID
+	tag                  TypeTag
```

Comment:
Mostly to encourage using the getter, and make it read-only as much as possible. (e.g: storing and reconstructing the tag can have disastrous effects, since they can be changed)

Reply:
Ah, I see 👍